### PR TITLE
Youtube - New Source "New Comment Posted"

### LINKED ISSUE #5179: [TRIGGER] YouTube new comment and reply
**Describe the event source. What app is this for, and what event does the trigger correspond to?**
YouTube new comment and reply

**Please provide a link to the relevant API docs for the specific service / operation this trigger is tied to.**
https://developers.google.com/youtube/v3/docs/comments/list

---
Would like a YouTube comment Triger and action as well.. when a comment is added in the channel then something, and also when something happens then reply with a comment on YouTube.

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #5179 

**Source**
- New Comment Posted

The actions "Create Comment Thread" and "Reply to Comment" are added in [this PR](https://github.com/PipedreamHQ/pipedream/pull/5371).

Note: I had to use the Youtube_Date_API_Custom_App to test this source due to the Youtube_Data_API app requiring an additional scope.

### CODE REVIEW

File: components/youtube_data_api/sources/new-comment-posted/common.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export default {
+  ...common,
+  hooks: {
+    ...common.hooks,
+    deploy() {},
+  },
+  methods: {
+    ...common.methods,
+    async listCommentThreads(params) {
```
Comment (andrewjschuang) [inline components/youtube_data_api/sources/new-comment-posted/common.mjs:11]: Can this method be in the app file?

File: components/youtube_data_api/sources/new-comment-posted/common.mjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    let maxPublishedAfter = publishedAfter || 0;
+
+    const params = {
+      part: "id,replies,snippet",
+      videoId: this.videoId,
+    };
+
+    const items = [];
+    while (true) {
+      const { data } = await this.listCommentThreads(params);
```
Comment (andrewjschuang) [inline components/youtube_data_api/sources/new-comment-posted/common.mjs:34]:
```suggestion
      const { data } = await this.youtube.listCommentThreads(params);
```

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
Thanks for the submission, I've left a few comments.

**Review by andrewjschuang — DISMISSED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed using Youtube_Date_API_Custom_App

@dannyroosevelt : Additional scope is required to use Youtube_Data_API

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**